hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -328,6 +341,43 @@ public ResponseCodeEnum adjustBalance(AccountID aId, TokenID tId, long adjustmen
 		return sanityChecked(aId, tId, token -> tryAdjustment(aId, tId, adjustment));
 	}
 
+	@Override
+	public ResponseCodeEnum adjustBalance(AccountID senderAId, AccountID receiverAId, TokenID tId, long serialNumber) {
```

Comment:
Actually, where do we validate that the `senderAId` is the current owner of this `serialNumber`?